src/algebra/big_operators/fin.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+import algebra.big_operators.basic
+/-!
+# Big operators and `fin`
+
+Some results about products and sums over the type `fin`.
+-/
+
+open_locale big_operators
+
+open finset
+
+namespace fin
+
+@[simp]
+lemma univ_filter_zero_lt {n : ℕ} :
+  (univ : finset (fin n.succ)).filter (λ i, 0 < i) =
+    univ.map ⟨_, succ_injective _⟩ :=
+begin
+  ext i,
+  simp only [mem_filter, mem_map, mem_univ, true_and,
+    function.embedding.coe_fn_mk, exists_true_left],
+  split,
+  { refine cases _ _ i,
+    { rintro ⟨⟨⟩⟩ },
+    { intros i _, exact ⟨i, mem_univ _, rfl⟩ } },
+  { rintro ⟨i, _, rfl⟩,
+    exact succ_pos _ },
+end
+
+@[simp]
+lemma univ_filter_succ_lt {n : ℕ} (j : fin n) :
+  (univ : finset (fin n.succ)).filter (λ i, j.succ < i) =
+    (univ.filter (λ i, j < i)).map ⟨_, succ_injective _⟩ :=
+begin
+  ext i,
+  simp only [mem_filter, mem_map, mem_univ, true_and,
+      function.embedding.coe_fn_mk, exists_true_left],
+  split,
+  { refine cases _ _ i,
+    { rintro ⟨⟨⟩⟩ },
+    { intros i hi,
+      exact ⟨i, mem_filter.mpr ⟨mem_univ _, succ_lt_succ_iff.mp hi⟩, rfl⟩ } },
+  { rintro ⟨i, hi, rfl⟩,
+    exact succ_lt_succ_iff.mpr (mem_filter.mp hi).2 },
+end
```

Comment:
While these are useful for big operators, I think they belong somewhere else: perhaps `data.fintype.basic`?

Reply:
`data.fintype.basic` is quite long - perhaps `data.fintype.fin` would be a good idea?